### PR TITLE
Documentation

### PR DESCRIPTION
**Pull request by SamHellawell**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,10 +1,47 @@
...
+
+Roadmap:
+- Support requesting specific credentials/data
+- Decouple verification from the Dock API
+- Look into supporting the SIOP spec
+- OpenID Connect implementation
+- Support more DID types
+- Support non self-signed credentials
+
+## As an oauth2 provider
```
Comment (mike-parkhill) [inline README.md:22]: I think the preferred usage is `OAuth 2.0` in documentation

File: README.md [hunk 2]
```diff
@@ -1,10 +1,47 @@
...
+
+This service can be used directly as an oauth2 provider with your favourite oauth library. See the documentation for [Oauth2 setup](docs/oauth2_setup.md). You are welcome to use our hosted version or your own. Setup instructions are for the hosted vesion, simply replace with your own domain to configure for another endpoint.
+
+## As an Auth0 connection
```
Comment (mike-parkhill) [inline README.md:26]: remove Auth0 reference for now

File: README.md [hunk 3]
```diff
@@ -1,10 +1,47 @@
...
+
+## Under the Hood
+
+Decentralized Identifiers (DIDs) are cryptographically verifiable pseudonymous identifiers created by the user, owned by the user, and independent of any organization. DIDs contain no personal data about the user, the user may provide extra data you request such as their name, email etc. An example of a DID stored on the Dock blockchain could look like this:
+
+![sample-did](./public/DID%20example.jpg)
+
+Each DID is supported by a Public-Private cryptographic key pair.
+
+When a user scans the QR Code generated by the Dock Web3 ID Auth0 integration, the user’s Private Key associated with the DID digitally signs a Verifiable Credential. This Verifiable Credential contains a cryptographic hash that ensures that it wasn’t modified since it was created and signed - and most importantly verifies that the user who owns that DID is providing that data.
```
Comment (mike-parkhill) [inline README.md:38]: remove Auth0 reference here - we probably don't need it even when we do bring Auth0 integration out.

File: docs/auth0_configure_application.md [hunk 4]
```diff
@@ -2,24 +2,8 @@
 
 The Dock Web3 ID integration allows users to privately sign in to your app using [Decentralized Identifiers (DIDs)](https://docs.api.dock.io/#dids) & [Verifiable Credentials (VCs)](https://www.dock.io/) via scanning QR codes with their Dock Wallet apps. This step-by-step guide will show you how to integrate Dock Web3 ID into your authentication flow easily.
 
-## Under the Hood
```
Comment (mike-parkhill) [inline docs/auth0_configure_application.md:5]: I understand why you moved this, but this document will also be the source of the `Installation` page on the Auth0 marketplace so we need to keep the context for there.

File: README.md (hunk 2 above)
Comment (SamHellawell) [reply README.md:26]: i will be removing auth0 in separate PR for easy revert

File: README.md (hunk 3 above)
Comment (SamHellawell) [reply README.md:38]: see other comment

File: docs/auth0_configure_application.md (hunk 4 above)
Comment (SamHellawell) [reply docs/auth0_configure_application.md:5]: i think that we can just copy/paste from the initial readme too when we come to adding that document as im sure theres more context (like the intro paragraph) that would be useful also

**Review by mike-parkhill — APPROVED** [review]
Your plan to remove the Auth0 references in a separate PR makes sense

**Pull request merged**